comp/process/bundle_test.go

```diff
@@ -6,32 +6,46 @@
 package process
 
 import (
-	"context"
 	"testing"
 
+	"github.com/stretchr/testify/mock"
 	"github.com/stretchr/testify/require"
 	"go.uber.org/fx"
 
-	"github.com/DataDog/datadog-agent/comp/core"
-	"github.com/DataDog/datadog-agent/comp/core/config"
-	"github.com/DataDog/datadog-agent/comp/core/sysprobeconfig"
+	sysconfig "github.com/DataDog/datadog-agent/cmd/system-probe/config"
 	"github.com/DataDog/datadog-agent/comp/process/runner"
-	"github.com/DataDog/datadog-agent/comp/process/submitter"
+	"github.com/DataDog/datadog-agent/pkg/process/checks"
+	checkMocks "github.com/DataDog/datadog-agent/pkg/process/checks/mocks"
 	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
 )
 
+func newMockCheck(t testing.TB, name string) *checkMocks.Check {
+	// TODO: Change this to use check componenet once checks are migrated
+	mockCheck := checkMocks.NewCheck(t)
+	mockCheck.On("Init", mock.Anything, mock.Anything).Return(nil)
+	mockCheck.On("Name").Return(name)
+	mockCheck.On("SupportsRunOptions").Return(false)
+	mockCheck.On("Realtime").Return(false)
+	mockCheck.On("Cleanup")
+	mockCheck.On("Run", mock.Anything, mock.Anything).Return(&checks.StandardRunResult{}, nil)
+	mockCheck.On("ShouldSaveLastRun").Return(false)
+	mockCheck.On("IsEnabled").Return(true)
+	return mockCheck
+}
+
 func TestBundleDependencies(t *testing.T) {
 	require.NoError(t, fx.ValidateApp(
-		fx.Supply(fx.Annotate(t, fx.As(new(testing.TB)))),
+		fx.Supply(
+			fx.Annotate(t, fx.As(new(testing.TB))),
+
+			[]checks.Check{checkMocks.NewCheck(t)},
+			&checks.HostInfo{},
+			&sysconfig.Config{},
+		),
 
 		// instantiate all of the process components, since this is not done
 		// automatically.
 		fx.Invoke(func(r runner.Component) {}),
-		fx.Invoke(func(r submitter.Component) {}),
```

Comment:
As a heads up, I will add invocations for these components as I build them.

For now, I want to be able to control the components that are constructed, as some of the downstream dependants of components in the process bundle that I haven't touched yet change global state when they are constructed.